substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java

```diff
@@ -129,6 +129,9 @@ protected SystemPropertiesSupport() {
         initializeProperty("java.endorsed.dirs", "");
         initializeProperty("java.ext.dirs", "");
         initializeProperty("sun.arch.data.model", Integer.toString(ConfigurationValues.getTarget().wordJavaKind.getBitCount()));
+        // JDK's StaticProperty initializes the useOldISOCodes property to
+        // the empty string and Locale code relies on it being non-null.
+        initializeProperty("java.locale.useOldISOCodes", System.getProperty("java.locale.useOldISOCodes", ""));
```

Comment:
On second thought, I think we should instead fix the `jdk.internal.util.StaticProperty.javaLocaleUseOldISOCodes()` substitution to return `SystemPropertiesSupport.singleton().savedProperties.getOrDefault("java.locale.useOldISOCodes", "")`, see https://github.com/oracle/graal/blob/9df286d2ac54eecbc66170c08e76ce23bcda9d34/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_jdk_internal_util_StaticProperty.java#L142

WDYT?

Reply:
That works as well. Updated.

Reply:
Turns out that @jovanstevanovic stumbled upon the same problem recently and is going to fix this with his next PR.

Reply:
Fair enough. Please close #9526 once that PR merges.

Reply:
I'm going to merge this to unblock Quarkus :)

Reply:
That would be fantastic! :)